controllers/common.go

```diff
@@ -599,6 +770,37 @@ func (r *MultiClusterHubReconciler) waitForMCEReady(ctx context.Context) (ctrl.R
 	return ctrl.Result{}, nil
 }
 
+func (r *MultiClusterHubReconciler) waitForHostedMCEReady(ctx context.Context, m *operatorv1.MultiClusterHub) (ctrl.Result, error) {
+	// Wait for MCE to be ready
+	existingMCE, err := multiclusterengine.GetHostedManagedMCE(ctx, r.Client, m)
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+	if existingMCE == nil {
+		r.Log.Info(fmt.Sprintf("Multiclusterengine: %s is not yet present", existingMCE.GetName()))
```

Comment:
When running this code it panicked here from calling GetName() of nil and couldn't progress further.